tests/python/contrib/test_hexagon/benchmark_elemwise_add.py

```diff
@@ -0,0 +1,468 @@
+            f.write("LOWERED IR MODULE:\n")
+            f.write(str(module_for_ir_dump))
+            f.write("\n")
+
+            target_hexagon = tvm.target.hexagon("v68", link_params=True)
+            func = tvm.build(
+                # sched,
+                ir_module,
+                [A, B, C],
+                tvm.target.Target(target_hexagon, host=target_hexagon),
+                name="elemwise_add",
+            )
+
+            host_dso_binary_path = os.path.join(host_files_dir, "test_binary.so")
+            target_dso_binary_filename = "test_binary.so"
+
+            func.save(host_dso_binary_path)
+            print(f"SAVED BINARY TO HOST PATH: {host_dso_binary_path}")
+
+            hexagon_launcher.upload(host_dso_binary_path, target_dso_binary_filename)
+
+            try:
+                with hexagon_launcher.start_session() as sess:
+                    mod = hexagon_launcher.load_module(target_dso_binary_filename, sess)
+
+                    host_numpy_A_data = np.ndarray(shape, dtype=dtype)
+                    host_numpy_B_data = np.ndarray(shape, dtype=dtype)
+
+                    for i in range(shape[0]):
+                        for j in range(shape[1]):
+                            host_numpy_A_data[i, j] = i + j
+                            host_numpy_B_data[i, j] = (i + 1) * (j + 1)
+
+                    host_numpy_C_data_expected = host_numpy_A_data + host_numpy_B_data
+
+                    A_data = tvm.nd.empty(shape, dtype, sess.device, mem_scope)
+                    A_data.copyfrom(host_numpy_A_data)
+
+                    B_data = tvm.nd.empty(shape, dtype, sess.device, mem_scope)
+                    B_data.copyfrom(host_numpy_B_data)
+
+                    C_data = tvm.nd.empty(shape, dtype, sess.device, mem_scope)
+
+                    # NOTE: We may want to soften these numbers, depending on future findings.
+                    timer = mod.time_evaluator("main", sess.device, number=10, repeat=1)
+                    timing_result = timer(A_data, B_data, C_data)
+
+                    print("TIMING RESULT: {}".format(timing_result))
+
+                    # Verify that the computation actually happened, and produced the correct result.
+                    result = C_data.numpy()
+                    tvm.testing.assert_allclose(host_numpy_C_data_expected, result)
+
+                    _BT.record_success(timing_result, host_files_dir=host_files_dir, **keys_dict)
+
+            except Exception as err:
+                print()
+                print(f"FAILURE: See {report_path}")
+                f.write("ERROR:\n")
+                f.write(f"{err}\n")
+                _BT.record_fail(
+                    **keys_dict, host_files_dir=host_files_dir, comments=f"See {report_path}"
+                )
+
+    # -----------------------------------------------------------------------------------------------
+
+    # Hexagon v69 allows more dtypes, but we're sticking with v68 for now.
+    for dtype in [
```

Comment:
nit:
These loops can be hoisted out into `tvm.testing.parameters`, e.g. 

```
class TestEWBenchmarks:
    dtype = tvm.testing.parameter("int8")
    num_vectors_per_tensor = tvm.testing.parameters([1, 16, 64, 512])
    ...
    @tvm.testing.requires_hexagon
    def test_elemwise_add_tvmcript(hexagon_session : HexagonLauncherRPC, dtype,  num_vectors_per_tensor, ...):

```

In doing so, each testing option becomes a separate pytest. The benchmarking utilities could be defined in the class scope to capture the state across multiple tests.

Reply:
It's a good idea.  If possible I'd like to tackle that in a future PR, once we have more benchmark configurations written.